### ISSUE TITLE
Allow Debug Logging of Logits

### ISSUE THREAD

**Issue by lapp0** [dottxt-ai/outlines#614]
### What behavior of the library made you think about the improvement?

If the model generates bad results we currently have no idea why. E.g.
- https://github.com/outlines-dev/outlines/issues/612#issuecomment-1928014413
- https://github.com/outlines-dev/outlines/issues/580#issuecomment-1909236093

### How would you like it to behave?

For each generation step, we should have an optional debug log of
- 1) The softmaxed logits and decoded representation of the top N tokens
- 2) The softmaxed logits of all special tokens (namely EOS)

**Comment by cpfiffer** [comment]
Related https://github.com/lapp0/outlines/pull/35
